specs/MultiProfile.md

```diff
@@ -0,0 +1,240 @@
+    {
+        ShowFailure(hr, L"Invalid profile name, create WebView failed! Window is closing.");
+        CloseAppWindow();
+        return S_FALSE;
+    }
+    CHECK_FAILURE(hr);
+
+#ifdef USE_WEBVIEW2_WIN10
+    if (webViewEnvironment4 && (m_dcompDevice || m_wincompCompositor))
+#else
+    if (webViewEnvironment4 && m_dcompDevice)
+#endif
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2CompositionControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2CompositionControllerCompletedHandler>(
+                [this](
+                    HRESULT result,
+                    ICoreWebView2CompositionController* compositionController) -> HRESULT {
+                    auto controller =
+                        wil::com_ptr<ICoreWebView2CompositionController>(compositionController)
+                            .query<ICoreWebView2Controller>();
+                    return OnCreateCoreWebView2ControllerCompleted(result, controller.get());
+                })
+                .Get()));
+    }
+    else
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2ControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2ControllerCompletedHandler>(
+                this, &AppWindow::OnCreateCoreWebView2ControllerCompleted)
+                .Get()));
+    }
+
+    return S_OK;
+}
+```
+
+## Access the profile property of WebView2
+
+### Win32 C++
+
+```cpp
+HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICoreWebView2Controller* controller)
+{
+    // ...
+
+    m_controller = controller;
+    
+    // Gets the webview object from controller.
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+    coreWebView2.query_to(&m_webView);
+    m_webView3 = coreWebView2.try_query<ICoreWebView2_3>();
+    
+    // Gets the profile property of webview.
+    wil::com_ptr<ICoreWebView2Profile> profile;
+    CHECK_FAILURE(m_webView3->get_Profile(&profile));
+    
+    // Accesses the profile object.
+    wil::unique_cotaskmem_string name;
+    CHECK_FAILURE(profile->get_ProfileName(&name));
+    BOOL inPrivateModeEnabled;
+    CHECK_FAILURE(profile->get_InPrivateModeEnablede(&inPrivateModeEnabled));
+    wil::unique_cotaskmem_string path;
+    CHECK_FAILURE(profile->get_ProfilePath(&path));
+    
+    // ...
+}
+```
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2ControllerOptions;
+interface ICoreWebView2Environment5;
+interface ICoreWebView2_7;
+interface ICoreWebView2Profile;
+
+[uuid(C2669A3A-03A9-45E9-97EA-03CD55E5DC03), object, pointer_default(unique)]
+interface ICoreWebView2ControllerOptions : IUnknown {
+  /// `ProfileName` property is to specify a profile name, which is only allowed to contain
+  /// the following ASCII characters with the maximum length as 64 and will be treated in a
```

Comment:
Is that 64 characters including the null terminator or excluding the null terminator?

Reply:
I think normally it should be excluding the null terminator. Will update the comment to make it clear.